### PR TITLE
Fix tranquiliser shells not working

### LINKED ISSUE #23807: Tranquilizer shots dont work
## Description

Shotting someone with tranquilizer shells dont affect them at all, even when shot by 8 in a row. Armor or no armor.
**Reproduction**

Shoot someone with tranquilizer rounds. No effect

### PR DESCRIPTION
**Pull request by nikthechampiongr**
Resolves #23807

## About the PR

Tranquilizers work again.

## Technical details

Changed the check for whether the fixture was hard to a check to see whether the fixture is the one that's supposed to trigger it. The fixtureid is saved in the component. Projectiles are apparently never hard.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/32041239/c79648a2-24e8-4493-847e-3c696d550562

**Changelog**

:cl:
- fix: Nanotrasen has fixed a recent design defect with tranquilizer shells. They will incapacitate people once more.

### CODE REVIEW

File: Content.Server/Chemistry/EntitySystems/SolutionInjectOnCollideSystem.cs [hunk 1]
```diff
@@ -26,7 +27,7 @@ private void HandleInjection(Entity<SolutionInjectOnCollideComponent> ent, ref S
         var target = args.OtherEntity;
 
         if (!args.OtherBody.Hard ||
-            !args.OurBody.Hard ||
+            args.OurFixtureId == FlyByFixture ||
             !EntityManager.TryGetComponent<BloodstreamComponent>(target, out var bloodstream) ||
```
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/SolutionInjectOnCollideSystem.cs:31]: SolutionInjectOnCollide needs to be checking a specific fixture instead.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by Arendian — COMMENTED** [review]
This will make it so hypodarts don't work ,because they don't have a projectile fixture. You should store a fixture on the component, and give the new fixture to the prototypes using InjectOnCollide that have no projectile fixture.

**Comment by nikthechampiongr** [comment]
> This will make it so hypodarts don't work ,because they don't have a projectile fixture. You should store a fixture on the component, and give the new fixture to the prototypes using InjectOnCollide that have no projectile fixture.

You are right I'll work on that. By the way unless you know of a way to get constants from systems in YAML I think this will create a bunch of constants that will need to be kept in sync manually which will suck.

**Comment by nikthechampiongr** [comment]
Should be good now. I tested it with the following. If I am missing something then please tell me. 
image

**Pull request merged**